src/common/types/value.cpp

```diff
@@ -281,8 +281,11 @@ std::string Value::toString() const {
         std::string result = "[";
         for (auto i = 0u; i < listVal.size(); ++i) {
             result += listVal[i]->toString();
-            result += (i == listVal.size() - 1 ? "]" : ",");
+            if (i != listVal.size() - 1) {
```

Comment:
I usually prefer `for i =0, I < size - 1` and write an additional line after the loop which avoids unnecessary runtime if checking. But up to u.

Reply:
if the listVal.size() is 0, i think this code: `for i =0, I < size - 1` will cause a bug.